guide/additional-features/cooldowns.md

```diff
@@ -0,0 +1,95 @@
+# Cooldowns
+
+Spam is something you generally want to avoid, especially if one of your commands require calls to other APIs or takes a bit of time to build/send.
+
+::: tip
+This section assumes you followed the [Command Handling](/creating-your-bot/command-handling.md) part.
+:::
+
+First, add a cooldown property to your command. This will determine how long the user would have to wait (in seconds) before using the command again.
+
+```js {4}
+const { SlashCommandBuilder } = require('discord.js');
+
+module.exports = {
+	cooldown: 5,
+	data: new SlashCommandBuilder()
+		.setName('ping')
+		.setDescription('Replies with Pong!'),
+	async execute(interaction) {
+		// ...
+	},
+};
+```
+
+In your main file, initialize a [Collection](/additional-info//collections.md) to store cooldowns of commands
+
+```js {1}
+client.cooldowns = new Collection();
+```
+
+The key will be the command names, and the values will be Collections associating the user's id (key) to the last time (value) this user used this command. Overall the logical path to get a user's last usage of a command will be `cooldowns > command > user > timestamp`.
+
+In your main file, add the following code:
+
+```js {1,3-5,7-10,12-14}
+const { cooldowns } = client;
+
+if (!cooldowns.has(command.data.name)) {
+	cooldowns.set(command.data.name, new Collection());
+}
+
+const now = Date.now();
+const timestamps = cooldowns.get(command.data.name);
+const defaultCooldownDuration = 3;
+const cooldownAmount = (command.cooldown || defaultCooldownDuration) * 1000;
+
+if (timestamps.has(interaction.user.id)) {
+	// ...
+}
+
+try {
+	// ...
+} catch (error) {
+	// ...
+}
+```
+
+You check if the `cooldowns` Collection already has an entry for the command being used. If this is not the case, you can add a new entry, where the value is initialized as an empty Collection. Next, create the following variables:
+
+1. `now`: The current timestamp.
+2. `timestamps`: A reference to the Collection of user ids and timestamp key/value pairs for the triggered command.
+3. `cooldownAmount`: The specified cooldown for the command, converted to milliseconds for straightforward calculation. If none is specified, this defaults to three seconds.
+
+If the user has already used this command in this session, get the timestamp, calculate the expiration time and inform the user of the amount of time they need to wait before using this command again. Note the use of `return` statement here, causing the code below this snippet to execute only if the user has not used this command in this session or the wait has already expired.
+
+Continuing with your current setup, this is the complete `if` statement:
+
+```js {2,4-7}
+if (timestamps.has(interaction.user.id)) {
+	const expirationTime = timestamps.get(interaction.user.id) + cooldownAmount;
+	if (now < expirationTime) {
+		const timeLeft = Math.round(expirationTime / 1000);
```

Comment:
Still its not *time left*
because its just unix timestamp

Reply:
unix timestamp of the time left, no?

Reply:
It’s the (absolute) timestamp of when the cooldown expires, not a relative duration. So `expirationTimestamp` would be more appropriate here